src/main/java/blockchains/iaas/uni/stuttgart/de/adaptation/AdapterManager.java

```diff
@@ -30,7 +27,7 @@
     private static final Logger log = LoggerFactory.getLogger(AdapterManager.class);
     private BlockchainAdapterFactory factory = new BlockchainAdapterFactory();
     private static AdapterManager instance = null;
-    private final Map<String, Pair<BlockchainAdapter, AbstractConnectionProfile>> map = Collections.synchronizedMap(new HashMap<>());
+    private final Map<String, Pair<BlockchainAdapter, Map<String, Object>>> map = Collections.synchronizedMap(new HashMap<>());
```

Comment:
Not valid anymore. I restored use of `AbstractConnectionProfile`. I had used `Map<String, Object>`  instead of `AbstractConnectionProfile` because previously application knew the classes extending `AbstractConnectionProfile` during compile time. So, `ObjectMapper` knew how to mapping the `connectionProfile` json object to specific instance of class extending `AbstractConnectionProfile`. So, when user uploaded connection profile say for e.g. through REST api, jersey app could find which instance of adapter it should create.

When we switched to plugin based approach, it is not possible for the app to know classes extending `AbstractConnectionProfile` at compile time. The connection profile classes are loaded at runtime using pf4j. So, object mapper could not map json to specific `AbstractConnectionProfile`. To overcome this issue, I though of pass `Map<String, Object>` to the plugin extension class and this extension class would take care of creating adapter after reading the needed properties. 

Current updated implementation: PR [link](https://github.com/akshay-ap/BlockchainAccessLayer/pull/2/files)
In above PR, application loads the connection profile class at runtime. Link [here](https://github.com/akshay-ap/BlockchainAccessLayer/blob/6ba706e738a832c12d0ba55e7b4cdb0c70871ed4/src/main/java/blockchains/iaas/uni/stuttgart/de/restapi/Controllers/PluginManagerController.java#L74).
Thus, we can use `AbstractConnectionProfile` again.